src/dbnode/persist/fs/files.go

```diff
@@ -1211,10 +1366,27 @@ func CommitLogsDirPath(prefix string) string {
 	return path.Join(prefix, commitLogsDirName)
 }
 
-// DataFileSetExistsAt determines whether data fileset files exist for the given namespace, shard, and block start.
-func DataFileSetExistsAt(filePathPrefix string, namespace ident.ID, shard uint32, blockStart time.Time) (bool, error) {
+// DataFileSetExists determines whether data fileset files exist for the given
+// namespace, shard, block start, and volume.
+func DataFileSetExists(
+	filePathPrefix string,
+	namespace ident.ID,
+	shard uint32,
+	blockStart time.Time,
+	volume int,
+) (bool, error) {
+	// This function is in some cases the bottleneck, thus, we use different
```

Comment:
This function can easily become a performance bottleneck if the implementation is slow or requires scanning directories with a large number of files in them (as is common if namespaces with long retentions are configured). As a result, instead of using existing helper function it implements an optimized code path only involves checking if a few specific files exist and contain the correct contents.

Reply:
👍 fixed a few typos here too.